### DISCUSSION TITLE
Do you move the dev dependency "typescript" into the root package.json?

### DISCUSSION THREAD

**Discussion by jtuchel** [vercel/turborepo#7210] [Help] [answered]
Since all my projects are using TS it would make sense to move this package to root level. And it seems to work fine.

Are there any concerns? Should I duplicate the dependency into each package.json instead?

Thanks in advance!

**Comment by anthonyshew** [comment, marked as answer]
In general, you want to install dependencies *where they are used*. It's important to undersatnd the tradeoffs:

- If you ever update the TypeScript version in the root, you will invalidate all the caches for all of your tasks in the entire repo.
- If you install TypeScript versions in each application, you have more granular caching and more flexibility in a larger scale repository. If you're concerned about keeping the versions in sync, `syncpack` is a great tool today for automating that syncing. We also want to build a `turbo install` feature in the future that makes this a built-in for Turborepos.

I'll be writing documentation soon on these tradeoffs and adding more detail to this discussion so stay tuned!
